di/src/test/java/com/interface21/beans/factory/support/DefaultListableBeanFactoryTest.java

```diff
@@ -185,4 +211,32 @@ public C(A a) {
             this.a = a;
         }
     }
+
+    @Configuration
+    public static class TestConfiguration {
+
+        private final CircularComponent circularComponent;
+
+        public TestConfiguration(CircularComponent circularComponent) {
+            this.circularComponent = circularComponent;
+        }
+
+        @Bean
+        public NormalComponent testBean() {
+            return new NormalComponent();
+        }
+    }
```

Comment:
```suggestion
    @Configuration
    public static class TestConfiguration {

        private final CircularComponent circularComponent;

        public TestConfiguration(CircularComponent circularComponent) {
            this.circularComponent = circularComponent;
        }

        @Bean
        public NormalComponent testBean() {
            return new NormalComponent();
        }

        @Bean
        public AbstractComponent firstBean() {
            return new FirstComponent();
        }

        @Bean
        public AbstractComponent secondBean() {
            return new SecondComponent();
        }
    }

    public interface AbstractComponent {}

    public static class FirstComponent implements AbstractComponent {}

    public static class SecondComponent implements AbstractComponent {}
```

위의 경우 처럼 같은 인터페이스 리턴타입 임에도, 
실제 리턴 인스턴스가 다른 경우에 대한 예외 테스트를 추가하면 좋을 것 같습니다.

Reply:
https://github.com/next-step/java-di/pull/12#discussion_r1703372957

에 답변달아두었습니다 :)